clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/util/id/DcosSpinnakerAppId.groovy

```diff
@@ -0,0 +1,153 @@
+
+        validateServerGroupName(this.marathonPath.last().get())
+    }
+
+    public DcosSpinnakerAppId(final String marathonAppId, final String accountName) {
+        this(marathonAppId)
+
+        checkArgument(this.account == accountName, "Account expected in DCOS Spinnaker App ID does not match given account name.")
+    }
+
+    public DcosSpinnakerAppId(final String account, final String region, final String serverGroupName) {
+        checkArgument(!nullToEmpty(account).trim().empty, "The account should not be null, empty, or blank.")
+        checkArgument(!nullToEmpty(region).trim().empty, "The region should not be null, empty, or blank.")
+        checkArgument(!nullToEmpty(serverGroupName).trim().empty, "The serverGroupName should not be null, empty, or blank.")
+
+        validateServerGroupName(serverGroupName)
+
+        marathonPath = MarathonPathId.parse("/${account}/${region.replaceAll(SAFE_REGION_SEPARATOR, MarathonPathId.PART_SEPARATOR)}/${serverGroupName}")
+    }
+
+    public String getAccount() {
+        marathonPath.first().get()
+    }
+
+    /**
+     * @return The canonical DC/OS "region" (a.k.a the full group path in which the marathon application lives)
+     *         including backslashes. This is returned as a relative path, meaning no preceeding backslash. Will never
+     *         be null.
+     *         <p/>
+     *         Deemed unsafe because various Spinnaker components have trouble with a region with backslashes.
+     *         <p/>
+     *         Ex: {@code foo/bar}
+     * @see #getSafeRegion()
+     */
+    public String getRegion() {
+        marathonPath.tail().parent().relative()
+    }
+
+    /**
+     * @return The canonical DC/OS "region" (a.k.a the full group path in which the marathon application lives)
+     *         including backslashes. This is returned as a relative path, meaning no preceeding backslash. Will never
+     *         be null.
+     *         <p/>
+     *         Deemed unsafe because various Spinnaker components have trouble with a region with backslashes.
+     *         <p/>
+     *         Ex: {@code foo/bar}
+     * @see #getSafeRegion()
+     */
+    public String getUnsafeRegion() {
+        region
+    }
+
+    /**
+     * @return The "safe" DC/OS region (a.k.a the group in which the marathon application lives). This is returned as a
+     *         relative path, meaning no preceeding underscore. Will never be null.
+     *         <p/>
+     *         Deemed safe because backslashes are replaced with underscores.
+     *         <p/>
+     *         Ex: {@code acct_foo_bar}
+     * @see #getSafeRegion()
+     */
+    public String getSafeRegion() {
+        region.replaceAll(MarathonPathId.PART_SEPARATOR, SAFE_REGION_SEPARATOR)
+    }
+
+    public Names getServerGroupName() {
+        Names.parseName(marathonPath.last().get())
+    }
+
+    public String getNamespace() {
```

Comment:
I searched the code and didn't see this getting used anywhere - can it be removed?